test/test_states/test_choices.js

```diff
@@ -450,9 +450,69 @@ describe("states.choice", function() {
 
+        describe("to make additional characters available", function() {
+            it("shouldn't count chars for .back on 1st pp and .more on last pp", function() {
+                opts.options_per_page = null;
+                opts.question = 'Select:',  // 8 chars (7 + \n)
+                opts.more = 'More',  // 8 chars (4 + 3 + \n)
+                opts.back = 'Back';  // 8 chars (4 + 3 + \n)
+                opts.characters_per_page = 24 + 16;  // should allow 2 off 8char choices
+
+                opts.choices = [
+                    new Choice('frut', 'Frut'), // 8 chars (4 + 3 + \n)
+                    new Choice('barz', 'Barz'), // 8 chars (4 + 3 + \n)
+                    new Choice('quux', 'Quux'), // 8 chars (4 + 3 + \n)
+                    new Choice('corg', 'Corg'), // 8 chars (4 + 3 + \n)
+                    new Choice('gnab', 'Gnab'), // 8 chars (4 + 3 + \n)
+                    new Choice('igni', 'Igni'), // 8 chars (4 + 3 + \n)
+                ];
+
+                return Q()
+                    .then(function() {
+                        return tester
+                            .start()
+                            .check.reply([
+                                "Select:",
+                                "1. Frut",
+                                "2. Barz",
+                                "3. Quux",
+                                "4. More"
+                            ].join('\n'))
+                            .check.reply.char_limit(opts.characters_per_page)
+                            .run();
+                    })
+                    .then(function() {
+                        return tester
+                            .inputs(null, '4')
+                            .check.reply([
+                                "Select:",
+                                "1. Corg",
+                                "2. Gnab",
+                                "3. Igni",
+                                "4. Back"
+                            ].join('\n'))
+                            .check.reply.char_limit(opts.characters_per_page)
+                            .run();
+                    })
+                    .then(function() {
+                        return tester
+                            .inputs(null, '4', '4')
+                            .check.reply([
+                                "Select:",
+                                "1. Frut",
+                                "2. Barz",
+                                "3. Quux",
+                                "4. More"
+                            ].join('\n'))
+                            .check.reply.char_limit(opts.characters_per_page)
+                            .run();
+                    });
+            });
+        });
+
         describe("when the options per page is not fixed", function() {
             it("should dynamically split the choices", function() {
-                opts.question = 'Hello.',
+                opts.question = 'Hello.';
```

Comment:
I changed this because it looked strange, but it doesn't seem to make any difference.